libraries/apollo-debug-server/src/commonMain/kotlin/com/apollographql/apollo3/debugserver/ApolloDebugServer.kt

```diff
@@ -3,26 +3,34 @@ package com.apollographql.apollo3.debugserver
 import com.apollographql.apollo3.ApolloClient
 import com.apollographql.apollo3.debugserver.internal.server.Server
 import com.apollographql.apollo3.debugserver.internal.server.createServer
+import okio.withLock
+import java.util.concurrent.locks.ReentrantLock
 
 object ApolloDebugServer {
   private val apolloClients = mutableMapOf<ApolloClient, String>()
   private var server: Server? = null
+  private val lock = ReentrantLock()
 
   fun registerApolloClient(apolloClient: ApolloClient, id: String = "client") {
-    if (apolloClients.containsKey(apolloClient)) error("Client '$apolloClient' already registered")
-    if (apolloClients.containsValue(id)) error("Name '$id' already registered")
-    apolloClients[apolloClient] = id
-    startOrStopServer()
+    lock.withLock {
```

Comment:
I don't think that works? This locks the writer but readers from the HTTP request thread won't use the lock and might iterate the list while it is modified

Reply:
Ah you're right! I liked the `AtomicReference` idea but didn't like that it's JVM only. But maybe that's ok.

Reply:
[AtomicFu](https://github.com/Kotlin/kotlinx-atomicfu) has KMP versions.

Reply:
Ah great! 🎉 
I think I'll use JVM `AtomicReference` to avoid adding the AtomicFu dependency, but with the knowledge that we can use it to support more targets.